components/nexus-repository/src/main/java/org/sonatype/nexus/repository/cache/CacheController.java

```diff
@@ -42,9 +42,11 @@ public CacheController(final int contentMaxAgeMinutes, @Nullable final String ca
   /**
    * After invoking this method, all {@link #isStale(CacheInfo)} checks will return true that has {@link CacheInfo} not
    * carrying same token as created in this method.
+   *
+   * @return the new cache Token that is in-effect after this method returns.
    */
-  public void invalidateCache() {
-    this.cacheToken = newCacheToken();
+  public String invalidateCache() {
+    return this.cacheToken = newCacheToken();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)